exit-challenger/src/tx-submitter/index.js

```diff
@@ -0,0 +1,135 @@
+const Web3 = require("web3");
+const asyncRedis = require("async-redis");
+const config = require("../../config/default.json");
+
+const { rootChainProvider } = config;
+const { exitHandlerAddress } = config;
+const web3 = new Web3(rootChainProvider);
+const password = process.env.PASSWORD;
+
+const client = asyncRedis.createClient({
+  host: config.redis.host,
+  port: config.redis.port
+});
+
+// Rewrite with tracking nonce
+let wallet;
+// encrypt wallet
+const encrypted = require("./keystore/keystore");
+
+if (Object.prototype.toString.call(encrypted) === "[object Object]") {
+  const accountObj = web3.eth.accounts.decrypt(encrypted, password);
+  web3.eth.accounts.wallet.add(accountObj);
+  wallet = web3.eth.accounts.wallet;
+} else {
+  throw Error("Exit-challenger cannot work without ethereum address");
+}
+
+async function txSubmitter(_task) {
+  // _task = {challengeMsgData: msgData, exitsCallData: callData}
+  const accountForSend = wallet[0];
+  // Tracking nonce
+  let nonce;
+  const nonceFromBlockhain = await web3.eth.getTransactionCount(
+    accountForSend.address
+  );
+  let nonceInfo = await client.get("nonceInfo");
+  if (!nonceInfo) {
+    nonceInfo = await client.set("nonceInfo", true);
+    await client.set("nonce", nonceFromBlockhain);
+  }
+  let nonceFromDB = await client.get("nonce");
+  nonceFromDB = parseInt(nonceFromDB, 10);
```

Comment:
would this work? This way we get get rid on `nonceInfo`

```suggestion
  let nonceFromDB = await client.get("nonce");
  if (!nonceInfo) {
     await client.set("nonce", nonceFromBlockhain);
     nonceFromDB = nonceFromBlockhain;
  }
  nonceFromDB = parseInt(nonceFromDB, 10);
```

Reply:
yes. it would work.